corgidrp/data.py

```diff
@@ -340,27 +344,31 @@ def __init__(self, data_or_filepath, pri_hdr=None, ext_hdr=None, err = None, dq
         else:
             # data has been passed in directly
             # creation of a new file in DRP eyes
+            if isinstance(data_or_filepath, float):
+                self.data = np.array([data_or_filepath])
+                if err is not None and isinstance(err, float):
+                    self.err = np.array([err])
+                else:
+                    self.err = np.array([0])
+            else:
+                self.data = data_or_filepath
+                if err is not None:
+                    if np.shape(self.data) != np.shape(err)[-self.data.ndim:]:
+                        raise ValueError("The shape of err is {0} while we are expecting shape {1}".format(err.shape[-self.data.ndim:], self.data.shape))
+                    #we want to have a 3 dim error array
+                    if err.ndim == self.data.ndim + 1:
+                        self.err = err
+                    else:
+                        self.err = err.reshape((1,)+err.shape)
```

Comment:
I think these lines of code (362, 364) produce an error of default datatype int, instead of float. Please change this is the KGAIN_ER should output only floats.